### PR TITLE
xmr-stak-cpu: init at 1.3.0

### PR DESCRIPTION
**Pull request by nico202**
###### Motivation for this change
Package missing. It's an alt-coin (monero) pool miner, so I don't know if it suits here or in misc.
Also, I don't know how to ensure that the donationLevel parameter is a double. I'm passing it as a string for now

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
Its looks for its configuration in the same directory

```
$ find /nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/
/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/
/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/bin
/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/bin/xmr-stak-cpu
/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/bin/config.txt
$ xmr-stak-cpu
[2017-10-17 17:04:39] : Failed to open config file config.txt.
```

This needs to be fixed otherwise `config.txt` cannot be changed.

**Comment by nico202** [comment]
It looks for in the current directory, so it's just a matter of copying it 

Il 17 ottobre 2017 18:49:02 CEST, "Jörg Thalheim" <notifications@github.com> ha scritto:
>Its looks for its configuration in the same directory
>
>```
>$ find /nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/
>/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/
>/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/bin
>/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/bin/xmr-stak-cpu
>/nix/store/9wmy1qpw8qrp94s0mvh3rdbpxi1gz68k-xmr-stak-cpu-1.3.0/bin/config.txt
>$ xmr-stak-cpu
>[2017-10-17 17:04:39] : Failed to open config file config.txt.
>```
>
>This needs to be fixed otherwise `config.txt` cannot be changed.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/30511#issuecomment-337281429

-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by Mic92** [comment]
At the least `config.txt` should not be in `$out/bin`.

File: pkgs/applications/altcoins/xmr-stak-cpu.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "v${version}-1.5.0";
+    sha256 = "1i12zdzkvmnmx2r40hp71vr019ws7s1gkabwj4my27z6hinw6f8b";
+  };
+
+  meta = {
+    description = "Universal Stratum pool miner (CPU version)";
+    homepage = https://github.com/fireice-uk/xmr-stak-cpu;
+    maintainers = [stdenv.lib.maintainers.nico202];
+    inherit version;
+  };
```
Comment (vyp) [inline pkgs/applications/altcoins/xmr-stak-cpu.nix]: Please add the license to meta: `license = stdenv.lib.licenses.gpl3;`.

File: pkgs/applications/altcoins/xmr-stak-cpu.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "xmr-stak-cpu";
+    rev = "v${version}-1.5.0";
+    sha256 = "1i12zdzkvmnmx2r40hp71vr019ws7s1gkabwj4my27z6hinw6f8b";
+  };
+
+  meta = {
+    description = "Universal Stratum pool miner (CPU version)";
+    homepage = https://github.com/fireice-uk/xmr-stak-cpu;
+    maintainers = [stdenv.lib.maintainers.nico202];
+    inherit version;
```
Comment (vyp) [inline pkgs/applications/altcoins/xmr-stak-cpu.nix]: This line is not needed.

**Review by vyp — CHANGES_REQUESTED** [review]

File: pkgs/applications/altcoins/xmr-stak-cpu.nix (hunk 2 above)
Comment (nico202) [reply pkgs/applications/altcoins/xmr-stak-cpu.nix]: Removed

File: pkgs/applications/altcoins/xmr-stak-cpu.nix (hunk 1 above)
Comment (nico202) [reply pkgs/applications/altcoins/xmr-stak-cpu.nix]: Added

**Comment by nico202** [comment]
Placed the config under etc/xmr-stak-cpu.config.txt. I don't know if keeping it config.txt is okay, or if I need to rename it to template-config.txt or else

**Comment by nico202** [comment]
Closing since they moved to a new repo with more architecture (cuda/opencl...) here: https://github.com/fireice-uk/xmr-stak

**Pull request closed without merging**